relayer/relayer.go

```diff
@@ -158,63 +160,61 @@ func NewRelayer(
 	return &r, sub, nil
 }
 
+// Helper to process missed blocks. Chooses as the starting block the maximum of the latest processed block and the configured start block height.
+// startBlockHeight cannot be nil
 func (r *Relayer) processMissedBlocks(
 	sub vms.Subscriber,
+	startBlockHeight *big.Int,
 ) error {
-	// Get the latest processed block height from the database.
+	if startBlockHeight == nil {
+		return fmt.Errorf("startBlockHeight cannot be nil")
+	}
+	// Attempt to get the latest processed block height from the database.
+	// Note that the retrieved latest processed block may have already been partially (or fully) processed by the relayer on a previous run. When
+	// processing a warp message in real time, which is when we update the latest processed block in the database, we have no way of knowing
+	// if that is the last warp message in the block
 	latestProcessedBlockData, err := r.db.Get(r.sourceBlockchainID, []byte(database.LatestProcessedBlockKey))
 
-	// The following cases are treated as successful:
+	// First, determine the height to process from. There are two cases:
 	// 1) The database contains the latest processed block data for the chain
-	//    - In this case, we parse the block height and process warp logs from that height to the current block
+	//    - In this case, we process from the maximum of the latest processed block and the configured start block height to the latest block
 	// 2) The database has been configured for the chain, but does not contain the latest processed block data
-	//    - In this case, we save the current block height in the database, but do not process any historical warp logs
+	//    - In this case, we process from the start block height to the latest block
+	height := startBlockHeight
 	if err == nil {
-		// If the database contains the latest processed block data, then back-process all warp messages from that block to the latest block
-		// Note that the retrieved latest processed block may have already been partially (or fully) processed by the relayer on a previous run. When
-		// processing a warp message in real time, which is when we update the latest processed block in the database, we have no way of knowing
-		// if that is the last warp message in the block
+		// Use the max of the latest processed block and the start block height
 		latestProcessedBlock, success := new(big.Int).SetString(string(latestProcessedBlockData), 10)
 		if !success {
 			r.logger.Error("failed to convert latest block to big.Int", zap.Error(err))
 			return err
 		}
+		height = utils.MaxBigInt(latestProcessedBlock, height)
```

Comment:
can we add a log here about using latest process block height from db

Reply:
Do you feel strongly about logging the case when we pull the height from the DB specifically? We already log the height that we're processing from [here](https://github.com/ava-labs/awm-relayer/blob/008e978175449f32f24b35b8c6430e11e7365dd0/vms/evm/subscriber.go#L139), and to add the logging as described we wouldn't be able to use `utils.MaxBigInt`

Reply:
Same as below, I meant a log that just tells user we are using latest value from db instead of starting block height config value.

Reply:
Can be debug too, whichever you think is mroe appropriate

Reply:
Gotcha. Added.